internal/oci/container.go

```diff
@@ -448,13 +456,18 @@ func (c *Container) verifyPid() error {
 }
 
 // getPidStartTime reads the kernel's /proc entry for stime for PID.
-func getPidStartTime(pid int) (int, error) {
-	var st unix.Stat_t
-	if err := unix.Stat(fmt.Sprintf("/proc/%d", pid), &st); err != nil {
-		return 0, errors.Wrapf(ErrNotFound, err.Error())
+func getPidStartTime(pid int) (string, error) {
+	statContents, err := ioutil.ReadFile(fmt.Sprintf("/proc/%d/stat", pid))
+	if err != nil {
+		return "", errors.Wrapf(ErrNotFound, err.Error())
+	}
+
+	statFields := strings.Fields(string(statContents))
```

Comment:
1. `strings.Fields` is a slow function. Using `strings.Split` is preferable.

2. Field 2 can contain spaces, so it's not that easy to parse; in fact I think it's not possible as there can be any characters in there:

```console
$ cp /bin/sleep 'ha ha ))) '
$ ./ha\ ha\ \)\)\)\  1h &
[1] 7397
$ cut -c-50 /proc/7397/stat
7397 (ha ha ))) ) S 3859 7397 3859 34816 7412 1077
```

In case it's a user-controlled process (and I believe it is), it can change its name freely.


3. Since you do not need all the fields, it is probably better to employ a faster algo -- get past the field 2 (which is cumbersome), skip a certain number of spaces, get the value.

Reply:
Alternatively, we could also retrieve the program name for the pid in a different command and strip it away. But this seems slower in that case. :shrug:

Reply:
luckily, it's actually fairly simple, because the command is always wrapped in a set of parenthesis